File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnAsset.java

package com.gracenote.gnsdk;

/** 
* Assets for content (cover art, biography etc) 
* <p> 
* <p><b>SDK Documentation</b></p> 
* <p> 
* Topic:  Data Models 
* <p> 
*  The above topic displays all the metadata returned from a recognition event, 
*  including objects containing a {@link GnAsset} object. 
*  It does this in an expand/collapse format, showing object hierarchy and 
*  a description of each field returned. 
* <p> 
*/ 
 
public class GnAsset extends GnDataObject {
  private transient long swigCPtr;

  protected GnAsset(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnAsset_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnAsset obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnAsset(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

/** 
*  Asset dimension 
*  @return Dimention string 
*/ 
 
  public String dimension() {
    return gnsdk_javaJNI.GnAsset_dimension(swigCPtr, this);
  }

/** 
*  Asset Type 
*  @return asset's mime type string 
*/ 
 
  public String type() {
    return gnsdk_javaJNI.GnAsset_type(swigCPtr, this);
  }

  public int bytes() {
    return gnsdk_javaJNI.GnAsset_bytes(swigCPtr, this);
  }

/** 
*  Pixel image size of asset as defined with a {@link GnImageSize} enum, such as kImageSize110 (110x110) 
*  @return Image size 
*/ 
 
  public GnImageSize size() {
    return GnImageSize.swigToEnum(gnsdk_javaJNI.GnAsset_size(swigCPtr, this));
  }

/** 
* @deprecated NB: This method has been marked as deprecated and will be removed from the next major release. 
*      Use UrlHttp() or UrlHttps() instead. 
*/ 
 
  public String url() {
    return gnsdk_javaJNI.GnAsset_url(swigCPtr, this);
  }

/** 
*  GNSDK url for retrieval of asset from Gracenote 
*  @return URL 
*/ 
 
  public String urlGnsdk() {
    return gnsdk_javaJNI.GnAsset_urlGnsdk(swigCPtr, this);
  }

/** 
*  HTTP url for retrieval of asset from Gracenote 
*  @return URL 
*/ 
 
  public String urlHttp() {
    return gnsdk_javaJNI.GnAsset_urlHttp(swigCPtr, this);
  }

/** 
*  HTTPS url for retrieval of asset from Gracenote 
*  @return URL 
*/ 
 
  public String urlHttps() {
    return gnsdk_javaJNI.GnAsset_urlHttps(swigCPtr, this);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnAsset.cs

namespace GracenoteSDK {

/**
* Assets for content (cover art, biography etc)
*
* <p><b>SDK Documentation</b></p>
*
* Topic:  Data Models
*
*  The above topic displays all the metadata returned from a recognition event,
*  including objects containing a GnAsset object.
*  It does this in an expand/collapse format, showing object hierarchy and
*  a description of each field returned.
*
*/
public class GnAsset : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnAsset(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnAsset_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnAsset obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnAsset() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnAsset(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

/**
*  Asset Type
*  @return asset's mime type string
*/
  public string Type() {
	System.IntPtr temp = gnsdk_csharp_marshalPINVOKE.GnAsset_Type(swigCPtr); 
	return GnMarshalUTF8.StringFromNativeUtf8(temp);
}

/**
*  GNSDK url for retrieval of asset from Gracenote
*  @return URL
*/
  public string UrlGnsdk() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnAsset_UrlGnsdk(swigCPtr);
    return ret;
  }

/**
*  HTTP url for retrieval of asset from Gracenote
*  @return URL
*/
  public string UrlHttp() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnAsset_UrlHttp(swigCPtr);
    return ret;
  }

/**
*  HTTPS url for retrieval of asset from Gracenote
*  @return URL
*/
  public string UrlHttps() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnAsset_UrlHttps(swigCPtr);
    return ret;
  }

/**
*  Pixel image size of asset as defined with a GnImageSize enum, such as kImageSize110 (110x110)
*  @return Image size
*/
  public GnImageSize Size {
    get {
      GnImageSize ret = (GnImageSize)gnsdk_csharp_marshalPINVOKE.GnAsset_Size_get(swigCPtr);
      return ret;
    } 
  }

  public uint Bytes {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnAsset_Bytes_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Asset dimension
*  @return Dimention string
*/
  public string Dimension {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnAsset_Dimension_get(swigCPtr) );
	} 

  }

/**
* @deprecated NB: This method has been marked as deprecated and will be removed from the next major release.
*      Use UrlHttp() or UrlHttps() instead.
*/
  public string Url {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnAsset_Url_get(swigCPtr) );
	} 

  }

}

}
